### DISCUSSION TITLE
Checks failed before Status Check timeout reached

### DISCUSSION THREAD

**Discussion by andrewcarlisle-signifyd** [community/community#48723] [Pull Requests] [answered]
Merge Queue Status Check Timeout

Bug

I have the merge queue set up such that the build check (Team City) has a 4 hour timeout before a status check timeout. Yet we still have builds pulled from the queue stating that the check has failed after as little as 15 minutes, while the build is still ongoing and hasn't failed. What is going on? How can a build check fail before the build has reported anything back?

**Comment by Anmol-Baranwal** [comment, marked as answer]
I don't really know since you haven't shared any repo for reference. 
This might solve your issue: https://blog.mergify.com/how-to-handle-checks-timeout-in-an-evolutive-merge-queue/

**Reply by andrewcarlisle-signifyd** [reply]
Thanks, I can't share the repo since it is private. The merge queue branch protection configs are set up as follows
![image](https://user-images.githubusercontent.com/79850136/225696516-d5642cc3-5c91-4da4-881a-6ffda34511c7.png)
Builds will sometimes fail a build status check after 15 minutes even though the timeout is 4 hours and the build is still going and hasn't failed.

**Reply by Anmol-Baranwal** [reply]
1. Have you checked the limit on the size or duration of the build that you're hitting? For example, some CI/CD tools may have limits on the amount of time or resources that a build can consume.

To troubleshoot this issue further, you may want to review the logs and metrics from your build system to see if there are any clues as to why the build is failing prematurely. You may also want to try running the build locally to see if you can reproduce the issue and identify any potential causes. If you're still having trouble, it may be helpful to reach out to the support team for the build system or toolchain that you're using to see if they can provide any additional guidance or assistance.

**Reply by Anmol-Baranwal** [reply]
@andrewcarlisle-signifyd 

I just wanted to check in and see if you have any remaining questions or concerns about my answer. If not, would you please mark my response as accepted?

It would be greatly appreciated and it will also help other users who might be looking for a solution to a similar issue.

**Reply by andrewcarlisle-signifyd** [reply]
Thanks Anmol, the issue isn't that the build fails, the build doesn't fail it keeps running and passes on the CI tool. The issue is that the build gets kicked from the GH merge queue while the build is still running and hasn't had any issues. I thought it was the status timing out but that isn't the case. 

Are there any other reasons that a build could be kicked from the merge queue other than a status check failure or timeout? Thanks for your help.

**Reply by Anmol-Baranwal** [reply]
If the build is getting kicked from the GitHub merge queue while it is still running and hasn't failed or timed out, it is possible that there might be a network or connectivity issue. If the build tool or GitHub's servers experience irregular connectivity issues, it may cause the build to appear as failed or timed out, even though it is still running.

Another possibility is that there may be a problem with the configuration of the build tool or GitHub's integration with it. For example, there may be some configuration setting in the build tool that is causing the build to appear as failed or timed out prematurely, even though it is still running.

It is also possible that there may be some bug or issue with GitHub's merge queue itself that is causing the build to be kicked out prematurely.

You should check:

1. Check the network connectivity and make sure that the build tool and GitHub's servers are communicating without any issues.

2. Review the configuration settings for the build tool and GitHub's integration with it to ensure that there are no misconfigurations or settings that could be causing the issue.

3. Check the GitHub status page to see if there are any known issues with the merge queue or other GitHub services.

> Although, I am not sure if this could really fix the issue. 

Do let me know if the issue was fixed.

**Reply by andrewcarlisle-signifyd** [reply]
Thanks Anmol. The issue isn't fixed but it is very intermittent, happening a handful of time per week. It may be a connectivity issue, our build tool sometimes has issues collecting changes from Github so perhaps this is related. Thanks for your help, please consider the conversation resolved.

**Reply by Anmol-Baranwal** [reply]
Okay, thanks for letting me know.
